src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,44 @@
+package duke.task;
+
+/**
+ * Represents a task with status indicating its completeness
+ */
+public class Task {
```

Comment:
Maybe you can consider making Task an abstract class since it should not be instantiated?